cplex: download url for source package is outdated, package not compatible with newer versions

### Steps To Reproduce
Steps to reproduce the behavior:
1. Install pkgs.cplex.
2. Read error message.
3. Try to download https://developer.ibm.com/docloud/blog/2017/12/20/cplex-optimization-studio-12-8-now-available/
4. not reachable
5. Find different cplex version on IBM page, 22.1.0.
6. Try to build package with that cplex version

### Build log
```
building '/nix/store/918fqvjh5kx6bnq4wm5f2hd3m76q2wnr-cplex-128.drv'...
unpacking sources
patching sources
configuring
no configure script, doing nothing
building
JRE libraries are missing or not compatible....
Exiting....
glibPreInstallPhase
installing
dropIconThemeCache
glibPreFixupPhase
post-installation fixup
patchelf: getting info about '/nix/store/ax58hfnrw0wh11knmxbx4mgsfv3hdpxp-cplex-128/opl/bin/x86-64_linux/oplrun': No such file or directory
builder for '/nix/store/918fqvjh5kx6bnq4wm5f2hd3m76q2wnr-cplex-128.drv' failed with exit code 1
```

### Notify maintainers
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
Dear @bfortz I notice this package hasn’t been touched in a while, maybe you have moved on. But any hints how you would approach getting a working cplex, no matter the version, would be greatly appreciated.

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
system: "x86_64-linux"
host os: Linux 5.10.126, NixOS, 21.11 (Porcupine)
multi-user?: yes
sandbox: yes
version: nix-env (Nix) 2.3.16
channels(root): "nixos-21.11.337967.573603b7fdb, unstable-22.05pre379159.dfd82985c27"
nixpkgs: /nix/var/nix/profiles/per-user/root/channels/nixos
```

I am currently experiencing the same error. When using the latest version of CPLEX (which is, to my knowledge, 22.11), I get the following error:
```
~ nix-shell -p cplex           
this derivation will be built:
  /nix/store/mxf2wi28qvv61pr6b0ipssr884mzz6gc-cplex-128.drv
building '/nix/store/mxf2wi28qvv61pr6b0ipssr884mzz6gc-cplex-128.drv'...
Running phase: unpackPhase
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
Error: Could not find or load main class com.zerog.lax.LAX
Caused by: java.lang.ClassNotFoundException: com.zerog.lax.LAX
error: builder for '/nix/store/mxf2wi28qvv61pr6b0ipssr884mzz6gc-cplex-128.drv' failed with exit code 1;
       last 8 log lines:
       > Running phase: unpackPhase
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > no configure script, doing nothing
       > Running phase: buildPhase
       > Error: Could not find or load main class com.zerog.lax.LAX
       > Caused by: java.lang.ClassNotFoundException: com.zerog.lax.LAX
       For full logs, run 'nix log /nix/store/mxf2wi28qvv61pr6b0ipssr884mzz6gc-cplex-128.drv'.
```

It was working with CPLEX Version 20.10 on NixOS 23.11, but with the update to NixOS 24.05 I now get the below error. I guess this is due to an updated java dependency, which is incompatible with the CPLEX Version.
```
~ nix-shell -p cplex               
this derivation will be built:
  /nix/store/74rblwrqb245kkm4p525rahnzapfj6pl-cplex-128.drv
building '/nix/store/74rblwrqb245kkm4p525rahnzapfj6pl-cplex-128.drv'...
Running phase: unpackPhase
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
JRE libraries are missing or not compatible....
Exiting....
Running phase: glibPreInstallPhase
Running phase: installPhase
Running phase: dropIconThemeCache
Running phase: glibPreFixupPhase
Running phase: fixupPhase
patchelf: getting info about '/nix/store/ipr15pyg66wngwc88rrvlysq3w2mznm9-cplex-128/opl/bin/x86-64_linux/oplrun': No such file or directory
error: builder for '/nix/store/74rblwrqb245kkm4p525rahnzapfj6pl-cplex-128.drv' failed with exit code 1;
       last 10 log lines:
       > no configure script, doing nothing
       > Running phase: buildPhase
       > JRE libraries are missing or not compatible....
       > Exiting....
       > Running phase: glibPreInstallPhase
       > Running phase: installPhase
       > Running phase: dropIconThemeCache
       > Running phase: glibPreFixupPhase
       > Running phase: fixupPhase
       > patchelf: getting info about '/nix/store/ipr15pyg66wngwc88rrvlysq3w2mznm9-cplex-128/opl/bin/x86-64_linux/oplrun': No such file or directory
       For full logs, run 'nix log /nix/store/74rblwrqb245kkm4p525rahnzapfj6pl-cplex-128.drv'.
```

--- a/pkgs/applications/science/math/cplex/default.nix
+++ b/pkgs/applications/science/math/cplex/default.nix
@@ -9,14 +9,14 @@
 
 stdenv.mkDerivation rec {
   pname = "cplex";
-  version = "128";
+  version = "22.11";
 
   src =
     if releasePath == null then
       throw ''
         This nix expression requires that the cplex installer is already
         downloaded to your machine. Get it from IBM:
-        https://developer.ibm.com/docloud/blog/2017/12/20/cplex-optimization-studio-12-8-now-available/
+        https://www.ibm.com/support/pages/downloading-ibm-ilog-cplex-optimization-studio-2211
 
         Set `cplex.releasePath = /path/to/download;` in your
         ~/.config/nixpkgs/config.nix for `nix-*` commands, or
@@ -36,7 +36,8 @@ stdenv.mkDerivation rec {
   '';
 
   buildPhase = ''
-    sh $name -i silent -DLICENSE_ACCEPTED=TRUE -DUSER_INSTALL_DIR=$out
+     export JAVA_TOOL_OPTIONS=-Djdk.util.zip.disableZip64ExtraFieldValidation=true
+     sh $name LAX_VM ${openjdk}/bin/java -i silent -DLICENSE_ACCEPTED=TRUE -DUSER_INSTALL_DIR=$out
   '';
 
   installPhase = ''

